src/renderer/inc/RenderEngineBase.hpp

```diff
@@ -21,39 +21,91 @@ Author(s):
 #pragma once
 namespace Microsoft::Console::Render
 {
+    struct BufferLineRenderData
+    {
+        IRenderData* pData;

```

Comment:
Do you know what sort of stuff is left that is needed out of `IRenderData` beyond what you've already packed here? It feels like you have a lot of it already.

Reply:
I guess there isn't anything crucial left? I've managed to meet the minimum standard to not break the existing codebase., even though I don't understand some of them (for example, `LienRendition`, and `needLineTransformation`). I just blindly copied the old logic. With this PR, both conhost and WT run as usual. It may have glitches potentially, but at first glance this PR doesn't break the app entirely.